### DISCUSSION TITLE
[boost-iostreams] Should not depend on copy-left licensed code by default

### DISCUSSION THREAD

**Discussion by mschofie** [microsoft/vcpkg#19326] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Boost has a very specific license - [The Boost License](https://www.boost.org/LICENSE_1_0.txt). The "boost-iostreams" package, however, consumes ports with very different licenses _by default_. "boost-iostreams" (as per the [CONTROL file](https://github.com/microsoft/vcpkg/blob/master/ports/boost-iostreams/CONTROL)) depends on non-Boost ports that have other licenses. Specifically, liblzma and zstd have a mix of licenses that includes copy-left licenses. I don't think that consumers of the Boost library would expect copy-left dependencies included by default.

* liblzma - The [COPYING file](https://github.com/xz-mirror/xz/blob/master/COPYING) shows the multiple licenses that are applied. [ClearlyDefined.io declares the copyleft licenses](https://clearlydefined.io/definitions/git/github/xz-mirror/xz/b525b0c0ef40cd89b69294c9b8d57f4a8db58e1f).
* zstd - This project contains two licenses, for BSD-3 and GPL-2. [ClearlyDefined.io shows both](https://clearlydefined.io/definitions/git/github/facebook/zstd/10f0e6993f9d2f682da6d04aa2385b7d53cbb4ee).

**Proposed solution**
`boost-iostreams` should not depend on copy-left licensed code by default. Consumers should be able to opt-in.

**Describe alternatives you've considered**
Forking the ports is an option.

**Comment by PhoebeHui** [comment]
@mschofie, it seems the license https://github.com/boostorg/iostreams/blob/master/LICENSE is same with https://www.boost.org/LICENSE_1_0.txt, and it says 'boostorg/iostreams is licensed under the Boost Software License 1.0'.

@BillyONeal, could you help take a look?

**Comment by BillyONeal** [comment]
You would need to take this up with the boost-iostreams developers; boost-iostreams requires those external libraries. Boost makes claims about their own code, but clearly document that they need those external dependencies here: https://www.boost.org/doc/libs/1_74_0/libs/iostreams/doc/index.html

They do have some options to disable these compression features, however, they appear to default to enabling them, given the name of the supplied option being `NO_Xxx` rather than `USE_Xxx` or similar.

As a result I believe we should respect the upstream project's decision, and they turn those on by default.

We would certainly accept a patch to the port enabling those features to be disabled so that you could disable the problematic parts by saying something like `boost-iostreams[core]`.

**Comment by BillyONeal** [comment]
>clearly document that they need those external dependencies here

Well that link didn't work because they use frames :/

https://www.boost.org/doc/libs/1_74_0/libs/iostreams/doc/installation.html#ide

**Comment by mschofie** [comment]
Thanks, @BillyONeal, I appreciate the perspective. I agree, making a change to support opting out seems to be the right thing to do given the boost-iostream decisions.

**Comment by yurybura** [comment, marked as answer]
@mschofie @BillyONeal Starting from commit f904f7354dce4cb396a4d0792e36319d78ccccfb only the required dependencies can be installed together with `boost-iostreams`, all other will be disabled. This example enables only BZip2 and ZLIB:
```
vcpkg install boost-iostreams[core,bzip2,zlib]
```
